eva-accession-import-automation/generate_properties.py

```diff
@@ -0,0 +1,168 @@
+mongodb.read-preference=primaryPreferred
+
+spring.main.web-environment=false
+
+logging.level.uk.ac.ebi.eva.accession.dbsnp=DEBUG
+logging.level.uk.ac.ebi.eva.accession.dbsnp.listeners=INFO
+logging.level.org.springframework.jdbc.datasource=DEBUG
+"""
+    properties_filename = "{}.properties".format(args.assembly_accession)
+    with open(properties_filename, "w") as properties_file:
+        properties_file.write(template.format(args=args))
+
+
+def get_user_and_password_and_port_from_pgpass_for_host(pg_host):
+    with open(expanduser("~/.pgpass")) as pgpass_file:
+        for line in pgpass_file:
+            if line.split(':', 1)[0] == pg_host:
+                fields = line.rstrip('\n').split(':')
+                return fields[3], fields[4], fields[1]
+
+
+def init_logger():
+    logging.basicConfig(level=logging.INFO, format='%(asctime)-15s %(levelname)s %(message)s')
+    result_logger = logging.getLogger(__name__)
+    return result_logger
+
+
+if __name__ == "__main__":
+    logger = init_logger()
+
+    parser = argparse.ArgumentParser(
+        description='Generate a properties file that can be used for eva-accession-import',
+        add_help=False)
+    parser.add_argument("-b", "--build", help="dbSNP build number, e.g. 151", required=True)
+    parser.add_argument("-l", "--latest-build",
+                        help="Flag that this build is the latest (relevant for dbsnp table name)",
+                        action='store_true')
+    parser.add_argument("-n", "--assembly-name",
+                        help="Assembly name for which the process has to be run, e.g. Gallus_gallus-5.0"
+                             ". (Can be ommited if there is only one assembly name in the build)")
+    parser.add_argument("-a", "--assembly-accession",
+                        help="Assembly for which the process has to be run, e.g. GCA_000002315.3",
+                        required=True)
+    parser.add_argument("-r", "--assembly-report",
+                        help="File with GenBank equivalents for RefSeq accessions", required=True)
+    parser.add_argument("-f", "--fasta", help="FASTA file with the reference sequence",
+                        required=True)
+
+    parser.add_argument("-d", "--metadb", help="Postgres metadata DB", required=True)
+    parser.add_argument("-u", "--metauser", help="Postgres metadata DB username", required=True)
+    parser.add_argument("-h", "--metahost", help="Postgres metadata DB host", required=True)
+
+    parser.add_argument("-D", "--job-tracker-db", help="Postgres DB for the job repository",
+                        required=True)
+    parser.add_argument("-H", "--job-tracker-host", help="Postgres host for the job repository",
+                        required=True)
+
+    parser.add_argument("--mongo-db", help="MongoDB accessioning database", required=True)
+    parser.add_argument("--mongo-auth-db", help="MongoDB accessioning authentication database",
```

Comment:
I think it is just MongoDB authentication database

Reply:
that is just deployment details. if we had other mongo clusters the auth db could be different. this parameter is "the MongoDB authentication database for the accessioning DB"